lib/_http_client.js

```diff
@@ -346,15 +346,19 @@ function socketCloseListener() {
   // NOTE: It's important to get parser here, because it could be freed by
   // the `socketOnData`.
   var parser = socket.parser;
-  if (req.res && req.res.readable) {
-    // Socket closed before we emitted 'end' below.
-    req.res.emit('aborted');
-    var res = req.res;
-    res.on('end', function() {
+  var res = req.res;
+  if (res) {
+    if (res.readable) {
+      // Socket closed before we emitted 'end' below.
+      res.emit('aborted');
+      res.on('end', function() {
+        res.emit('close');
+      });
+      res.push(null);
+    } else {
       res.emit('close');
```

Comment:
Since `res.readable == false`, does that mean an `'end'` event has already been emitted?

Reply:
`end` will never be called if not readable at this point.